crates/ruff_python_formatter/src/pattern/pattern_match_as.rs

```diff
@@ -1,19 +1,38 @@
 use ruff_formatter::{write, Buffer, FormatResult};
 use ruff_python_ast::PatternMatchAs;
 
-use crate::{not_yet_implemented_custom_text, FormatNodeRule, PyFormatter};
+use crate::prelude::*;
+use crate::{FormatNodeRule, PyFormatter};
 
 #[derive(Default)]
 pub struct FormatPatternMatchAs;
 
 impl FormatNodeRule<PatternMatchAs> for FormatPatternMatchAs {
     fn fmt_fields(&self, item: &PatternMatchAs, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(
-            f,
-            [not_yet_implemented_custom_text(
-                "x as NOT_YET_IMPLEMENTED_PatternMatchAs",
-                item
-            )]
-        )
+        let PatternMatchAs {
+            range: _,
+            pattern,
+            name,
+        } = item;
+
+        if let Some(name) = name {
+            if let Some(pattern) = pattern {
+                write!(
+                    f,
+                    [
+                        pattern.format(),
+                        space(),
+                        text("as"),
+                        space(),
+                        name.format(),
+                    ]
+                )
+            } else {
+                name.format().fmt(f)
```

Comment:
Nit: We can move the `name` out of the branching
```suggestion
                    ]
                )?;
            } 
            name.format().fmt(f)?;
```